apps/gdalalg_raster_sieve.cpp

```diff
@@ -0,0 +1,245 @@
+        GDALTranslateOptionsFree(translateOptions);
+    }
+    else
+    {
+        // Translate the single required band to the destination format
+        CPLStringList options;
+        options.AddString("-b");
+        options.AddString(CPLSPrintf("%d", m_band));
+        options.AddString("-of");
+        options.AddString(m_format.c_str());
+        for (const auto &co : m_creationOptions)
+        {
+            if (co.find('=') != std::string::npos)
+            {
+                options.AddString(co.c_str());
+            }
+            else
+            {
+                options.AddString(CPLSPrintf("-co %s", co.c_str()));
+            }
+        }
+        GDALTranslateOptions *translateOptions =
+            GDALTranslateOptionsNew(options.List(), nullptr);
+        GDALDatasetH hSrcDS =
+            GDALDataset::ToHandle(m_inputDataset.GetDatasetRef());
+        workingDSHandler.reset(GDALDataset::FromHandle(GDALTranslate(
+            workingFileName.c_str(), hSrcDS, translateOptions, nullptr)));
+        GDALTranslateOptionsFree(translateOptions);
+    }
+
+    if (!workingDSHandler)
+    {
+        ReportError(CE_Failure, CPLE_AppDefined,
+                    "Failed to create temporary dataset");
+        return false;
+    }
+
+    GDALRasterBand *maskBand{nullptr};
+    if (m_maskDataset.GetDatasetRef())
+    {
+        maskBand = m_maskDataset.GetDatasetRef()->GetRasterBand(1);
+        if (!maskBand)
+        {
+            ReportError(CE_Failure, CPLE_AppDefined, "Cannot get mask band.");
+            return false;
+        }
+    }
+
+    // Get the output band
+    GDALRasterBand *dstBand{workingDSHandler->GetRasterBand(1)};
+
+    if (!dstBand)
+    {
+        ReportError(CE_Failure, CPLE_AppDefined,
+                    "Failed to get band %d from output dataset", m_band);
+        return false;
+    }
+
+    const CPLErr err = GDALSieveFilter(
+        dstBand, maskBand, dstBand, m_sizeThreshold,
+        m_connectDiagonalPixels ? 8 : 4, nullptr, pfnProgress, pProgressData);
+
+    if (err != CE_None)
+    {
+        ReportError(CE_Failure, CPLE_AppDefined,
+                    "Failed to apply sieve filter");
+        return false;
+    }
+
+    workingDSHandler->FlushCache();
+
+    // If we were using a temporary file we need to translate to the final desired format
+    if (useTemporaryFile)
+    {
+        // Translate the temporary dataset to the final format
+        CPLStringList options;
+        options.AddString("-b");
+        options.AddString(CPLSPrintf("%d", m_band));
+        options.AddString("-of");
+        options.AddString(m_format.c_str());
+        for (const auto &co : m_creationOptions)
+        {
+            if (co.find('=') != std::string::npos)
+            {
+                options.AddString(co.c_str());
+            }
+            else
+            {
+                options.AddString(CPLSPrintf("-co %s", co.c_str()));
+            }
```

Comment:
Test likely lacking for that code path:
```suggestion
           options.AddString("-co");
           options.AddString(co.c_str());
```